src/panoptes/pocs/focuser/serial.py

```diff
@@ -0,0 +1,180 @@
+    # Private Methods
+    ##################################################################################################
+
+    def _connect(self, port, baudrate):
+        try:
+            # Configure serial port.
+            self._serial_port = serial.Serial()
+            self._serial_port.port = port
+            self._serial_port.baudrate = baudrate
+            self._serial_port.bytesize = serial.EIGHTBITS
+            self._serial_port.parity = serial.PARITY_NONE
+            self._serial_port.stopbits = serial.STOPBITS_ONE
+            self._serial_port.timeout = 2.0
+            self._serial_port.xonxoff = False
+            self._serial_port.rtscts = False
+            self._serial_port.dsrdtr = False
+            self._serial_port.write_timeout = None
+            self._inter_byte_timeout = None
+
+            # Establish connection
+            self._serial_port.open()
+
+        except serial.SerialException as err:
+            self._serial_port = None
+            self.logger.critical('Could not open {}!'.format(port))
+            raise err
+
+        # Want to use a io.TextWrapper in order to have a readline() method with universal newlines
+        # (focuser adaptors usually send '\r', not '\n'). The line_buffering option causes an automatic flush() when
+        # a write contains a newline character.
+        self._serial_io = io.TextIOWrapper(io.BufferedRWPair(self._serial_port, self._serial_port),
+                                           newline='\r', encoding='ascii', line_buffering=True)
+        self.logger.debug('Established serial connection to {} on {}.'.format(self.name, port))
+
+    def _send_command(self, command, response_length=None, error_pattern=None, error_messages=[]):
+        """
+        Sends a command to the focuser adaptor and retrieves the response.
+
+        Args:
+            command (string): command string to send (without newline), e.g. 'fa1000', 'pf'
+            response length (integer, optional, default=None): number of lines of response expected.
+                For most commands this should be 0 or 1. If None readlines() will be called to
+                capture all responses. As this will block until the timeout expires it should only
+                be used if the number of lines expected is not known (e.g. 'ds' command).
+
+        Returns:
+            list: possibly empty list containing the '\r' terminated lines of the response from the
+                adaptor.
+        """
+        if not self.is_connected:
+            self.logger.critical("Attempt to send command to {} when not connected!".format(self))
+            return
+
+        # Clear the input buffer in case there's anything left over in there.
+        self._serial_port.reset_input_buffer()
+
+        # Send command
+        self._serial_io.write(command + '\r')
+
+        if self.model == "astromechanics":
+            return self._serial_io.readline()
```

Comment:
Is everything below specific to a birger?  Seems like the more generic case is to just do the read and return and the more specific case should be handled where it is needed.